### PR TITLE
Add Docker build instructions

### LINKED ISSUE #73: Try add locally docker running way
I try to add locally docker running way.

I change the Makefile like below.
```
DOCKER       = docker
HUGO_VERSION = 0.58.0
DOCKER_IMAGE = tikv-hugo
DOCKER_RUN   = $(DOCKER) run --rm --interactive --tty --volume $(CURDIR):/src
NODE_BIN     = node_modules/.bin
NETLIFY_FUNC = $(NODE_BIN)/netlify-lambda

serve:
	hugo server \
		--buildDrafts \
		--buildFuture \
		--disableFastRender \
		--bind 0.0.0.0

serve-production:
	hugo server \
		--disableFastRender \
		--buildFuture \
		--bind 0.0.0.0

production-build:
	hugo --minify \
		--buildFuture

preview-build:
	hugo \
		--buildDrafts \
		--buildFuture \
		--baseURL $(DEPLOY_PRIME_URL) \
		--minify

docker-image:
	$(DOCKER) build . --tag $(DOCKER_IMAGE) --build-arg HUGO_VERSION=$(HUGO_VERSION)

docker-build:
	$(DOCKER_RUN) $(DOCKER_IMAGE) hugo

docker-serve:
	$(DOCKER_RUN) -p 13131:13131 $(DOCKER_IMAGE) hugo server --buildDrafts --buildFuture --disableFastRender --bind 0.0.0.0

```

And add the Dockerfile to root folder.
```
# Credit to Julien Guyomard (https://github.com/jguyomard). This Dockerfile
# is essentially based on his Dockerfile at
# https://github.com/jguyomard/docker-hugo/blob/master/Dockerfile. The only significant
# change is that the Hugo version is now an overridable argument rather than a fixed
# environment variable.

FROM alpine:latest

MAINTAINER ChengChen <cc8602@gmail.com>

RUN apk add --no-cache \
    curl \
    git \
    openssh-client \
    rsync \
    build-base \
    libc6-compat

ARG HUGO_VERSION

RUN mkdir -p /usr/local/src && \
    cd /usr/local/src && \
    curl -L https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_extended_${HUGO_VERSION}_Linux-64bit.tar.gz | tar -xz && \
    mv hugo /usr/local/bin/hugo && \
    addgroup -Sg 1000 hugo && \
    adduser -Sg hugo -u 1000 -h /src hugo

WORKDIR /src

EXPOSE 13131
```

But it failed to startup.

I export the debug information.
```
$ make docker-serve
docker run --rm --interactive --tty --volume /home/chengchen/go/src/github.com/tikv-fork/website:/src -p 13131:13131 tikv-hugo hugo server --buildDrafts --buildFuture --disableFastRender --bind 0.0.0.0 --debug
INFO 2019/09/18 00:19:37 No translation bundle found for default language "en"
INFO 2019/09/18 00:19:37 Translation func for language en not found, use default.
INFO 2019/09/18 00:19:37 i18n not initialized; if you need string translations, check that you have a bundle in /i18n that matches the site language or the default language.
DEBUG 2019/09/18 00:19:37 Add template file: name "_default/blog/card.html", baseTemplatePath "", path "_default/blog/card.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "_default/blog/navbar-link.html", baseTemplatePath "", path "_default/blog/navbar-link.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "_default/single.html", baseTemplatePath "_default/baseof.html", path "_default/single.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "blog/list.html", baseTemplatePath "_default/baseof.html", path "blog/list.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "blog/single.html", baseTemplatePath "_default/baseof.html", path "blog/single.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "docs/list.html", baseTemplatePath "_default/baseof.html", path "docs/list.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "docs/single.html", baseTemplatePath "_default/baseof.html", path "docs/single.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "index.html", baseTemplatePath "_default/baseof.html", path "index.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "index.redirects", baseTemplatePath "", path "index.redirects"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/admonition.html", baseTemplatePath "", path "partials/admonition.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/blog/hero.html", baseTemplatePath "", path "partials/blog/hero.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/blog/post-list.html", baseTemplatePath "", path "partials/blog/post-list.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/blog/post.html", baseTemplatePath "", path "partials/blog/post.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/css.html", baseTemplatePath "", path "partials/css.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/docs/hero.html", baseTemplatePath "", path "partials/docs/hero.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/docs/version-warning.html", baseTemplatePath "", path "partials/docs/version-warning.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/entry-tree.html", baseTemplatePath "", path "partials/entry-tree.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/footer.html", baseTemplatePath "", path "partials/footer.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/google-analytics.html", baseTemplatePath "", path "partials/google-analytics.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/javascript.html", baseTemplatePath "", path "partials/javascript.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/math.html", baseTemplatePath "", path "partials/math.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/meta.html", baseTemplatePath "", path "partials/meta.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/navbar.html", baseTemplatePath "", path "partials/navbar.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/open-graph-tags.html", baseTemplatePath "", path "partials/open-graph-tags.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "partials/twitter-card.html", baseTemplatePath "", path "partials/twitter-card.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/adopters.html", baseTemplatePath "", path "shortcodes/adopters.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/config.html", baseTemplatePath "", path "shortcodes/config.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/danger.html", baseTemplatePath "", path "shortcodes/danger.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/features.html", baseTemplatePath "", path "shortcodes/features.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/figure.html", baseTemplatePath "", path "shortcodes/figure.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/info.html", baseTemplatePath "", path "shortcodes/info.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/success.html", baseTemplatePath "", path "shortcodes/success.html"
DEBUG 2019/09/18 00:19:37 Add template file: name "shortcodes/warning.html", baseTemplatePath "", path "shortcodes/warning.html"
INFO 2019/09/18 00:19:37 Using config file: 
Building sites … INFO 2019/09/18 00:19:37 syncing static files to /
WARN 2019/09/18 00:19:37 found no layout file for "HTML" for "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
ERROR 2019/09/18 00:19:37 Transformation failed: SCSS processing failed: file "stdin", line 12, col 1: File to import not found or unreadable: bulma/sass/utilities/initial-variables. 
DEBUG 2019/09/18 00:19:37 found menu: "nav", in site config
DEBUG 2019/09/18 00:19:37 found menu: "nav", in site config
DEBUG 2019/09/18 00:19:37 found menu: "nav", in site config
DEBUG 2019/09/18 00:19:37 found menu: "nav", in site config
WARN 2019/09/18 00:19:37 found no layout file for "HTML" for "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
DEBUG 2019/09/18 00:19:37 Render XML for "sitemap" to "/sitemap.xml"
INFO 2019/09/18 00:19:37 found no layout file for "HTML" for "404": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
Total in 237 ms
Error: Error building site: logged 1 error(s)
Makefile:39: recipe for target 'docker-serve' failed
make: *** [docker-serve] Error 255
```

Do you know what happened?

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes #73 .

### CODE REVIEW

**Comment by Hoverbear** [comment]
Hi @it2911, I did a clean checkout of this repo and tried this dockerfile and run example, at the end I still owned the files it created. Can you help me understand the problem you faced?

**Comment by it2911** [comment]
@Hoverbear 

Thank you for confirming.
You have answered my questions in #76 ticket. I think it is no problem.
But the Dockerfile `USER builder` should have the role to create the folder else docker will fail run.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:latest
+RUN apk add hugo yarn make
```
Comment (it2911) [inline Dockerfile]: Before run `apk add hugo yarn make`, firstly run `apk update` is better.
The reason is adding component versions may be old. To keep the newest version is used should update repo first.

Comment (Hoverbear) [reply Dockerfile]: Great catch!

**Comment by Hoverbear** [comment]
Which folder isn't it able to create for you? I wonder if we need to do some uid tricks.

**Comment by Hoverbear** [comment]
@dcalvin Should be ready :)

**Pull request merged**